packages/types/src/payment/service.ts

```diff
@@ -1,3 +1,131 @@
+  ): Promise<PaymentCollectionDTO[]>
+  createPaymentCollection(
+    data: CreatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  retrievePaymentCollection(
+    paymentCollectionId: string,
+    config?: FindConfig<PaymentCollectionDTO>,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  listPaymentCollections(
+    filters?: FilterablePaymentCollectionProps,
+    config?: FindConfig<PaymentCollectionDTO>,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+
+  listAndCountPaymentCollections(
+    filters?: FilterablePaymentCollectionProps,
+    config?: FindConfig<PaymentCollectionDTO>,
+    sharedContext?: Context
+  ): Promise<[PaymentCollectionDTO[], number]>
+
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  deletePaymentCollection(
+    paymentCollectionId: string[],
+    sharedContext?: Context
+  ): Promise<void>
+  deletePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<void>
+
+  authorizePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  completePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  /* ********** PAYMENT ********** */
+
+  createPayment(data: CreatePaymentDTO): Promise<PaymentDTO>
+  createPayment(data: CreatePaymentDTO[]): Promise<PaymentDTO[]>
+
+  capturePayment(
+    paymentId: string,
+    amount: number,
+    sharedContext?: Context
+  ): Promise<PaymentDTO>
+  refundPayment(
+    paymentId: string,
+    amount: number,
+    sharedContext?: Context
+  ): Promise<PaymentDTO>
+
+  updatePayment(
+    data: UpdatePaymentDTO,
+    sharedContext?: Context
+  ): Promise<PaymentDTO>
+  updatePayment(
+    data: UpdatePaymentDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentDTO[]>
+
+  /* ********** PAYMENT SESSION ********** */
+
+  createPaymentSession(
+    paymentCollectionId: string,
+    data: CreatePaymentSessionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+  createPaymentSession(
+    paymentCollectionId: string,
+    data: CreatePaymentSessionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  authorizePaymentSessions(
+    paymentCollectionId: string,
+    sessionIds: string[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  completePaymentSessions(
+    paymentCollectionId: string,
+    sessionIds: string[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  setPaymentSessions(
+    paymentCollectionId: string,
+    data: SetPaymentSessionsDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
```

Comment:
suggestion: We should probably return the payment session instead of collection here right?

Reply:
The point of this method is to create/update/delete sessions on a PaymentCollection depending on passed data so I think it makes sense to return PaymentCollection object and its next set of sessions, wdyt? Btw, this is how we do it currently in the core payment collection service.